### ISSUE TITLE
Support a special intake catalog in data_stubs that is used when small data is needed.

### ISSUE THREAD

**Issue by jsignell** [holoviz-dev/pyct#71]
## Goal
Since we are now moving towards using intake rather than the custom datasets.yml, we'd like to have a set way of defining small data for intake. 

## First Attempt
The first example of this is in EarthML where we wrote a data/.data_stubs/catalog.yml with a subset of the sources defined in the regular catalog, but referring to smaller versions of the data. In EarthML there is a [python script](https://github.com/pyviz-topics/EarthML/blob/deb2ad7036b8f3bb3f0d6ad07f9c36b70d17d0e3/before_test.py#L1-L10)  that reads in the regular catalog from examples and the small one from data_stubs and writes the union of the two to a new catalog which then replaces the real catalog: https://github.com/pyviz-topics/EarthML/blob/master/dodo.py#L7-L12. 

This is not a great solution for a couple reasons:
 1) It relies on there being only one catalog.yml which is a local file.
 2) It doesn't clean up after itself. So when running the tests locally you will have a catalog.yml that should not get checked in. 

## Proposal
I'd prefer to have the path to the small catalog as an argument to nbsmoke and  nbsite so that on each call to open_catalog we instead open the catalog and the small catalog like `cat = intake.open_catalog(['./catalog.yml', './data/.data_stubs/catalog.yml'])`, but I am not sure how that would be implemented other than a text replace when running the notebook which seems brittle.

**Comment by jbednar** [comment]
I agree that it would be better not to leave a messed up catalog.yml, but I can't think of a good solution of the top of my head (and have to run right now).
